src/me.rs

```diff
@@ -154,18 +154,27 @@ fn full_search(x_lo: isize, x_hi: isize, y_lo: isize, y_hi: isize, blk_h: usize,
   }
 }
 
+// Adjust block offset such that entire block lies within frame boundaries
+fn adjust_bo(bo: &BlockOffset, fi: &FrameInvariants, blk_w: usize, blk_h: usize) -> BlockOffset {
+  BlockOffset {
+    x: (bo.x as isize).min(fi.w_in_b as isize - blk_w as isize / 4).max(0) as usize,
+    y: (bo.y as isize).min(fi.h_in_b as isize - blk_h as isize / 4).max(0) as usize
+  }
+}
+
 pub fn estimate_motion_ss4(
   fi: &FrameInvariants, fs: &FrameState, ref_idx: usize, bo: &BlockOffset
 ) -> Option<MotionVector> {
   if let Some(ref rec) = fi.rec_buffer.frames[ref_idx] {
+    let blk_w = 64;
```

Comment:
With this block size, estimate_motion_ss4() will be always called for a 64x64 SB, correct?
If so, I think putting some asserts to detect such condition would lessen any misuse of this function?

Reply:
Removed the hardcoded 64x64 size in this function and using a function parameter instead.

Reply:
Yes, I see that, thanks!

Reply:
So, now _ss4() and _ss2() are for ME on 1/4 resolution and 1/2 resolution, resp., for any input block sizes.

Reply:
Yes, it should now work for any block size, but haven't tested block sizes other 64x64 for 1/4 resolution and 32x32 for 1/2 resolution